### PR TITLE
Do not use env.yaml file anymore

### LINKED ISSUE #6046: Remove env.yaml
Get rid of the env.yaml file. We do not need to store namespaces anymore, and there is no other info that we're storing in this file, so it makes sense to get rid of it.
Doing this might also help with v2-v3 transition since v2 can handle cases where there is no env.yaml as compared to half-baked env.yaml

### Acceptance Criteria

- [ ] Make sure `odo dev` only takes namespace at the beginning of the command and does not change if namespace changes [Move this to a separate issue of it's own.]
- [ ] Remove any references to `env.yaml`

/kind user-story

### LINKED ISSUE #6056: DEBUG_PORT variable is injected into container by `odo dev`
/kind bug

## What versions of software are you using?

**Operating System:**

Fedora Linux

**Output of `odo version`:**

odo v3.0.0-beta3 (644d94cde)

## How did you run odo exactly?

```
$ odo init --devfile nodejs --starter nodejs-starter --name mycomponent
$ odo dev
```

```
$ kubectl exec deploy/mycomponent-app -- bash -c 'echo $DEBUG_PORT'
5858
```

## Actual behavior

An environment variable is injected into the container, by `odo dev`, when this env var is not defined in the devfile.

## Expected behavior

No variable should be injected automatically by `odo dev`. The DEBUG_PORT variable should be described in the devfile. 
See for an example https://registry.devfile.io/viewer/devfiles/Community+java-maven

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug

**What does this PR do / why we need it:**

This PR removes the concept of env.yaml file for `odo`. 

> The `envinfo.EnvSpecificInfo` structure now only contains information about devfile: devfilePath and devfileObj. 
>
> The `pkg/storage` package is still using this structure to get the devfileObj. A refactoring of this package would be useful to move the devfilePath and devfileObj to the Context structure directly and get rid of the EnvSpecificInfo structure, but is not part of this PR.

**Which issue(s) this PR fixes:**

Fixes #6046 
Fixes #6056 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/envinfo/envinfo.go [hunk 1]
```diff
@@ -93,225 +35,17 @@ func NewEnvSpecificInfo(envDir string) (*EnvSpecificInfo, error) {
 // newEnvSpecificInfo retrieves the env.yaml file, if it does not exist, we return a *BLANK* environment file.
 func newEnvSpecificInfo(envDir string, fs filesystem.Filesystem) (*EnvSpecificInfo, error) {
```
Comment (valaparthvi) [inline pkg/envinfo/envinfo.go:36]: Nit: You can move the contents of this function to NewEnvSpecificInfo.

File: pkg/odo/cli/dev/dev.go [hunk 2]
```diff
@@ -213,13 +184,7 @@ func (o *DevOptions) Run(ctx context.Context) (err error) {
 	}
 
 	// add odo-file-index.json path to .gitignore
```
Comment (valaparthvi) [inline pkg/odo/cli/dev/dev.go:186]:
```suggestion
	// add .odo dir path to .gitignore
```

File: pkg/odo/genericclioptions/resolve.go [hunk 3]
```diff
@@ -59,43 +58,9 @@ func (o *internalCxt) resolveProjectAndNamespace(cmdline cmdline.Cmdline) error
 // Or If app is set in envfile, return its value
 // Or if createAppIfNeeded, returns the default app name
 func resolveApp(cmdline cmdline.Cmdline, localConfiguration localConfigProvider.LocalConfigProvider, createAppIfNeeded bool) string {
```
Comment (valaparthvi) [inline pkg/odo/genericclioptions/resolve.go:60]: Why don't we get rid of this function? If not, at least the unused arguments.

**Review by valaparthvi — CHANGES_REQUESTED** [review]
Code lgtm, have a few suggestions.

File: pkg/devfile/adapters/kubernetes/utils/utils.go [hunk 4]
```diff
@@ -16,9 +14,6 @@ import (
 const (
 	// _envProjectsRoot is the env defined for project mount in a component container when component's mountSources=true
 	_envProjectsRoot = "PROJECTS_ROOT"
-
-	// _envDebugPort is the env defined in the runtime component container which holds the debug port for remote debugging
-	_envDebugPort = "DEBUG_PORT"
```
Comment (rm3l) [inline pkg/devfile/adapters/kubernetes/utils/utils.go:21]: Wouldn't it make more sense to handle #6056 in a separate PR (with a more specific test case, which I did not see here)? I don't really see the link between `env.yaml` and the Debug port env var here ;-)

**Comment by rm3l** [comment]
I've also linked this PR as closing #5984, as I think it is covered here.

File: pkg/devfile/adapters/kubernetes/utils/utils.go (hunk 4 above)
Comment (rm3l) [reply pkg/devfile/adapters/kubernetes/utils/utils.go:21]: Ah, I see: the debug port used to be stored in `env.yaml`, right? In this case, I think what's missing (unless I missed it) is a test case for https://github.com/redhat-developer/odo/issues/6056 to ensure `odo dev` does not inject a `DEBUG_PORT` env var into the container.

Comment (feloy) [reply pkg/devfile/adapters/kubernetes/utils/utils.go:21]: Yes, this is the reason why it is done here. Let me check if a test exists for this case

File: tests/integration/cmd_dev_debug_test.go [hunk 5]
```diff
@@ -53,6 +53,13 @@ var _ = Describe("odo dev debug command tests", func() {
 				// We are just using this to validate if nodejs agent is listening on the other side
 				helper.HttpWaitForWithStatus("http://"+ports["5858"], "WebSockets request was expected", 12, 5, 400)
 			})
+
+			// #6056
+			It("should not add a DEBUG_PORT variable to the container", func() {
+				podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+				stdout := commonVar.CliRunner.Exec(podName, commonVar.Project, "--", "sh", "-c", "echo ${DEBUG_PORT}")
```
Comment (rm3l) [inline tests/integration/cmd_dev_debug_test.go:60]:
```suggestion
				stdout := commonVar.CliRunner.Exec(podName, commonVar.Project, "--", "sh", "-c", "echo -n ${DEBUG_PORT}")
```

I guess the trailing newline added by `echo` might be captured into `stdout`..

**Review by rm3l — CHANGES_REQUESTED** [review]

**Review by rm3l — APPROVED** [review]

**Pull request merged**